### PR TITLE
Speed up the creation of attention mask

### PR DESCRIPTION
**Pull request by yuantailing**
Prefer to use the inplace variant of triu_/tril_ because they are faster than the out-of-place variants since torch 2.3.0 (https://github.com/pytorch/pytorch/pull/115013).

### CODE REVIEW

**Comment by ethanhe42** [comment]
generally, mask will be created inside transformer engine if `--use-mcore-models`
